File: 백준/Silver/13567. 로봇/로봇.py
dx=[1,0,-1,0]
dy=[0,1,0,-1]
dir=0
valid=True
x=0
y=0
M,n=map(int,input().split())
for _ in range(n):
    commands,num=input().split()
    if(commands=="MOVE"):
        x+=dx[dir]*int(num)
        y+=dy[dir]*int(num)
        if(x>M or x<0 or y>M or y<0):
            valid=False
    else:
        if(num=='0'):
            dir=(dir+1)%4
        else:
            dir=(dir+3)%4
if(valid):
    print(x,y)
else:
    print(-1)

File: 백준/Silver/13567. 로봇/로봇.kt
fun main()
{
    val br=System.`in`.bufferedReader()
    val bw=System.`out`.bufferedWriter()
    val (M,n)=br.readLine().trim().split(" ").map{it.toInt()}
    var x=0
    var y=0
    var dir=0
    val dx= intArrayOf(1,0,-1,0)
    val dy= intArrayOf(0,-1,0,1)
    var valid=true
    for(i in 1..n)
    {
        val (command, a)=br.readLine().trim().split(" ")
        val d=a.toInt()
        if(valid==false)continue
        else if(command=="MOVE")
        {
            val nextx=x+dx[dir]*d
            val nexty=y+dy[dir]*d
            if(0<=nextx && nextx<=M && 0<=nexty && nexty<=M)
            {
                x=nextx
                y=nexty
            }
            else valid=false
        }
        else
        {
            if(d==0) dir=(dir+3)%4
            else dir=(dir+1)%4
        }
    }
    if(valid) bw.write("$x $y")
    else bw.write("-1")
    bw.close()
}